app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CriteriaController < ApplicationController
+
+  def new
+    @assignment = Assignment.find(params[:assignment_id])
+    @criterion_type = params[:criterion_type]
+    @criterion = get_class(@criterion_type).new
```

Comment:
Rather than define the helper, like this, first check whether the param is `"FlexibleCriterion"` or `"RubricCriterion"` (make a constant list in this controller which contains these two strings), and then use the `constantize` method.